app/models/instance.rb

```diff
@@ -15,6 +15,9 @@ def find_tenant_by_host(host)
     end
   end
 
+  VALID_HOST_NAME_REGEX = /\A[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\z/i
+  validates :host, format: { with: VALID_HOST_NAME_REGEX }, length: { maximum: 255 }
```

Comment:
> A "name" (Net, Host, Gateway, or Domain name) is a text string up
   to 24 characters drawn from the alphabet (A-Z), digits (0-9), minus
   sign (-), and period (.).  Note that periods are only allowed when
   they serve to delimit components of "domain style names". (See
   RFC-921)

See RFC document